dotcom-rendering/src/web/components/Picture.tsx

```diff
@@ -15,13 +19,27 @@ type Props = {
 };
 
 type ResolutionType = 'hdpi' | 'mdpi';
+type Pixel = number;
+type Breakpoint = number;
+
+export type DesiredWidth = {
+	breakpoint: Breakpoint;
+	width: Pixel;
+};
```

Comment:
I think you can make this an inferred  and narrower type with the following:

```suggestion
type Breakpoint = typeof breakpoint[keyof typeof breakpoints];

export type DesiredWidth = {
	breakpoint: Breakpoint;
	width: Pixel;
};
```

Reply:
It's worth keeping in mind that I don't know what this new code does. Sure, I could go away and learn about this feature but there will be more people like me coming along.

It's also true that this might be a small thing to learn but I am a big fan of simple code with as few requirements to check the docs as possible. Each time we expect people to know about more advanced features of Typescript we raise the bar for access to DCR

Reply:
This code narrows the type: `number` &rarr; `320 | 640 | 960`.

I don’t find the syntax particularly elegant, and obviously you can create [a wrapper][] like `Values<T>`, but this will still fall within what you call advanced features of TypeScript. I don’t have a good answer for this, and it’s really up to the team to decide whether they prefer knowing which number is returned by a function. If you do, then the compiler can tell you if you’ve missed certain branches, which I like, but may not be everyone’s cup of tea.

[a wrapper]: https://www.typescriptlang.org/play?#code/MYewdgzgLgBARgJwKYEMDWAHEBLMUIwC8MA3gFAyUwC2Ic2ANkgFwwBsADBwDQVVQo4TKKwDsAFh59KAEyQQ0UEBlYBOTryowA7tjmsAjAGYumqgHoAVDEBkBDEvmyAXxgoCoSFADcZMlACeGEgwAGooDACu8gA8ACowSAAeUEhgMgQASkigCDLR0Ai4AObcMBFgaGAg2mAAfLVEMLEA2mhI-iAAZk0Aun6BwQBCyOhYuPiNYZExAUFd8COYOHgQtUA